python/cppyy/_cpython_cppyy.py

```diff
@@ -114,6 +114,22 @@ def __call__(self, *args):
       # most common cases are covered
         if args:
             args0 = args[0]
+
+            if (
+                type(args0).__module__ == "numpy"
+                and type(args0).__name__ == "ndarray"
+                and hasattr(args0, "dtype")
+            ):
+                t = args0.dtype.type.__name__
```

Comment:
> What I would suggest here is that if we want to drop the requirement for template args for std.vector

Not requiring template args for vector came about b/c since C++17 it's legal not to do that in C++. That is, something like this "`auto v = std::vector({1, 2, 3});`" is legal C++ and the argument was that Python should be at least as simple. (It isn't really, b/c C++ literals are a richer type set than Python literals, but resolving basic cases was/is simple enough.)